src/pages/study-page/study-results-page/study-results-page.tsx

```diff
@@ -92,4 +118,49 @@ export const StudyResultsPage = ({
       </>
     );
   }
+
+  async function handleFinishClick() {
+    if (isUpdating) {
+      return;
+    }
+
+    // update card scores
+    setIsUpdating(true);
+    await _updateAllCardScores();
+    setIsUpdating(false);
+
+    // navigation is handled with a side effect on `areResultsApplied`
+    setAreResultsApplied(true);
+  }
+
+  // TODO: it might be beneficial to have an endpoint to batch these (an array of {id, score})
+  // we are currently making requests equal to the number of seen lesson cards (marked correct/incorrect)
+  async function _updateAllCardScores() {
+    if (isUpdating) {
+      return;
+    }
+
+    const promises = [];
+
+    // create promise to update cards with new scores
+    for (const lessonCard of lessonCards) {
+      if (lessonCard.outcome === 'unseen') {
+        continue;
+      }
+
+      // spaced repetition: if correct, score is increased by 1; otherwise reset to 0
+      let newScore = 0;
+      if (lessonCard.outcome === 'correct') {
+        newScore = Math.min(lessonCard.score + 1, MAX_SCORE);
```

Comment:
Does the MAX_SCORE correspond with the number boxes in the spaced repetition system?

Reply:
it corresponds with the number of boxes - 1 since it follows 0-based indexing

Reply:
oh, haha. Yeah, that makes sense!